numpy/random/_examples/cython/extending_distributions.pyx

```diff
@@ -72,3 +74,46 @@ def uint10_uniforms(Py_ssize_t n):
     randoms = np.asarray(random_values)
     return randoms
 
+# cython example 3
+def uniforms_ex(Py_ssize_t n, dtype=np.float64):
+    """
+    Create an array of `n` uniformly distributed doubles via a "fill" function.
+
+    A 'real' distribution would want to process the values into
+    some non-uniform distribution
+
+    Parameters
+    ----------
+    n: int
+        Output vector length
+    dtype: {str, dtype}, optional
+        Desired dtype, either 'd' (or 'float64') or 'f' (or 'float32'). The
+        default dtype value is 'd'
+    """
+    cdef Py_ssize_t i
+    cdef bitgen_t *rng
+    cdef const char *capsule_name = "BitGenerator"
+    cdef np.ndarray randoms
+
+    typedict = {'f': np.float32, 'd': np.float64, 'float64': np.float64,
+                'float32': np.float32, np.float32: np.float32,
+                np.float64: np.float64}
+    typ = typedict.get(dtype, None)
+    if not typ:
+        raise TypeError('Unsupported dtype "%r"' % dtype)
+    x = PCG64()
+    capsule = x.capsule
+    # Optional check that the capsule if from a BitGenerator
+    if not PyCapsule_IsValid(capsule, capsule_name):
+        raise ValueError("Invalid pointer to anon_func_state")
+    # Cast the pointer
+    rng = <bitgen_t *> PyCapsule_GetPointer(capsule, capsule_name)
+    randoms = np.empty(n, dtype=dtype)
+    if typ is np.float32:
+        with x.lock:
+            random_standard_uniform_fill_f(rng, n, <float*>np.PyArray_DATA(randoms))
+    else:
+        with x.lock:
+            random_standard_uniform_fill(rng, n, <double*>np.PyArray_DATA(randoms))
+    return randoms
+
```

Comment:
@jamesthomasgriffin is this along the lines of what you are trying to do? It uses a new `pxd`, see line 13. Also the `setup.py` needs to link with the new `npyrandom` library.

Reply:
@mattip Yes that example would show me how to do what I wanted to do.  Thank you.